### ISSUE TITLE
Better configuration error

### ISSUE THREAD

**Issue by st0012** [carrierwaveuploader/carrierwave#1736]
Hello, I noticed that if we configure a not yet registered uploader like:
```
configure.storage = :foo
```

It will raise 
```
TypeError: no implicit conversion of nil into String
```
which is caused by `eval` in [this line](https://github.com/carrierwaveuploader/carrierwave/blob/master/lib%2Fcarrierwave%2Fuploader%2Fconfiguration.rb#L75)

And this is a very confusing message, you have to dive into source code to find out what's wrong. So I wonder if there anyone think raise a custom exception here will be a good idea?

**Comment by thomasfedb** [comment]
@st0012 Are you able to provide a pull request to improve the error message?

**Comment by st0012** [comment]
@thomasfedb Sure, I will work on it.

**Comment by bensie** [comment]
Closed by #1779
